src/soilbiogeochem/SoilBiogeochemNitrogenStateType.F90

```diff
@@ -711,10 +1001,45 @@ subroutine SetValues ( this, num_column, filter_column, value_column )
           do fi = 1,num_column
              i = filter_column(fi)
              this%decomp_npools_vr_col(i,j,k) = value_column
+             if(use_soil_matrixcn)then
+                this%matrix_cap_decomp_npools_vr_col(i,j,k) = value_column
+                this%decomp0_npools_vr_col(i,j,k) = value_column
+             end if
           end do
        end do
     end do
 
+    do j = 1,nlevdecomp
+       do k = 1, ndecomp_pools
+          do fi = 1, num_column
+             i = filter_column(fi)
+             if(use_soil_matrixcn)then
```

Comment:
The use_soil_matrixcn if should be put around the whole do loop rather than nested inside.

Reply:
done